R/call.R

```diff
@@ -49,9 +216,13 @@ call_new <- function(f, ..., .args = list()) {
 #'
 #' # Add an explicit missing argument
 #' call_modify(call, list(na.rm = quote(expr = )))
-call_modify <- function(call, new_args, env = parent.frame()) {
-  stopifnot(is.call(call), is.list(new_args))
-
+#'
+#' # If the call is missing, the parent frame is used instead.
+#' f <- function(bool = TRUE) call_modify(new_args = list(bool = FALSE))
+#' f()
+call_modify <- function(call = NULL, new_args, env = NULL) {
```

Comment:
It's a bit weird to have an optional arg, then a required arg, then an optional arg.